src/analysis/inner_product_space/projection.lean

```diff
@@ -862,6 +885,108 @@ lemma finrank_orthogonal_span_singleton {n : ℕ} [_i : fact (finrank 𝕜 E = n
+    have hW : ∀ w ∈ W, φ w = w := λ w hw, (sub_eq_zero.mp hw).symm,
+    by_cases hn' : finrank ℝ Wᗮ ≤ n,
+    { obtain ⟨V, hV₁, hV₂⟩ := IH φ hn',
+      exact ⟨V, hV₁.trans n.le_succ, hV₂⟩ },
+    -- Take a nonzero element `v` of the orthogonal complement of `W`.
+    haveI : nontrivial Wᗮ := nontrivial_of_finrank_pos (by linarith [zero_le n] : 0 < finrank ℝ Wᗮ),
+    obtain ⟨v, hv⟩ := exists_ne (0:Wᗮ),
+    have hφv : φ v ∈ Wᗮ,
+    { intros w hw,
+      rw [← hW w hw, linear_isometry_equiv.inner_map_map],
+      exact v.prop w hw },
+    have hv' : (v:F) ∉ W,
+    { intros h,
+      exact hv ((submodule.mem_left_iff_eq_zero_of_disjoint W.orthogonal_disjoint).mp h) },
+    -- Let `ρ` be the reflection in `v - φ v`; this is designed to swap `v` and `φ v`
+    let x : F := v - φ v,
+    let ρ := reflection (ℝ ∙ x)ᗮ,
+    -- Notation: Let `V` be the fixed subspace of `φ.trans ρ`
+    let V := (continuous_linear_map.id ℝ F - (φ.trans ρ).to_continuous_linear_equiv).ker,
+    have hV : ∀ w, ρ (φ w) = w → w ∈ V,
+    { intros w hw,
+      change w - ρ (φ w) = 0,
+      rw [sub_eq_zero, hw] },
+    -- Everything fixed by `φ` is fixed by `φ.trans ρ`
+    have H₂V : W ≤ V,
+    { intros w hw,
+      apply hV,
+      rw hW w hw,
+      refine reflection_mem_subspace_eq_self _,
+      apply mem_orthogonal_singleton_of_inner_left,
+      exact submodule.sub_mem _ v.prop hφv _ hw },
+    -- `v` is also fixed by `φ.trans ρ`
+    have H₁V : (v:F) ∈ V,
+    { apply hV,
+      have : ρ v = φ v := reflection_sub ↑v (φ v) (by simp),
+      simp [← this, ρ] },
+    -- By dimension-counting, the complement of the fixed subspace of `φ.trans ρ` has dimension at
+    -- most `n`
+    have : finrank ℝ Vᗮ ≤ n,
+    { change finrank ℝ Wᗮ ≤ n + 1 at hn,
+      have : finrank ℝ W + 1 ≤ finrank ℝ V :=
+        submodule.finrank_lt_finrank_of_lt (set_like.lt_iff_le_and_exists.2 ⟨H₂V, v, H₁V, hv'⟩),
+      have := @submodule.finrank_add_finrank_orthogonal _ _ _ _ _ V,
+      have := @submodule.finrank_add_finrank_orthogonal _ _ _ _ _ W,
+      linarith },
+    -- So apply the inductive hypothesis to `φ.trans ρ`
+    obtain ⟨l, hl, hφl⟩ := IH (φ.trans ρ) this,
+    -- Append `ρ` to the factorization into reflections obtained for `φ.trans ρ`; this gives a
+    -- factorization into reflections for `φ`.
+    refine ⟨(l.cons x), _, _⟩,
+    { simp [hl, nat.succ_le_succ] },
+    have := congr_arg (λ ψ, linear_isometry_equiv.trans ψ ρ) hφl,
+    convert this using 1,
+    { simp [← linear_isometry_equiv.trans_assoc φ ρ ρ] },
+    { change _ = _ * _,
+      simp } }
+end
+
+/-- The orthogonal group of `F` is generated by reflections; specifically each element `φ` of the
+orthogonal group is a product of at most as many reflections as the dimension of `F`. -/
+lemma linear_isometry_equiv.reflections_generate_dim [finite_dimensional ℝ F] (φ : F ≃ₗᵢ[ℝ] F) :
+  ∃ V : list F, V.length ≤ finrank ℝ F ∧ φ = (V.map (λ v, reflection (ℝ ∙ v)ᗮ)).prod :=
```

Comment:
Am I right in thinking that this is that "Cartan–Dieudonné theorem"?

Reply:
I'm pretty sure this theorem is far older than that -- the versions due to Cartan and Dieudonné appear to deal with more general signatures of symmetric form, more general scalar rings, etc.

Reply:
Would you be opposed to adding `a special case of the **Cartan–Dieudonné theorem**` to the docstring anyway?

Reply:
Done.